clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -11247,12 +11247,16 @@ void LinkerWrapper::ConstructJob(Compilation &C, const JobAction &JA,
         LinkerArgs.emplace_back("-lompdevice");
 
       // Forward all of these to the appropriate toolchain.
-      for (StringRef Arg : CompilerArgs)
-        CmdArgs.push_back(Args.MakeArgString(
-            "--device-compiler=" + TC->getTripleString() + "=" + Arg));
-      for (StringRef Arg : LinkerArgs)
-        CmdArgs.push_back(Args.MakeArgString(
-            "--device-linker=" + TC->getTripleString() + "=" + Arg));
+      // SYCL offload toolchains handle device compiler and linker argument forwarding using SYCLImage.
+      // Therefore, we skip forwarding these arguments here when a SYCL offload toolchain is present.
+      if (!C.hasOffloadToolChain<Action::OFK_SYCL>()) {
```

Comment:
Is it possible to avoid this SYCL-specific customization? What happens if we keep this for SYCL as well?
If I understand correctly @mdtoguchi : "The expected behavior is for any device options to be handled at both compile and link time. ", we need to keep this code and pass user-specified options to clang-linker-wrapper same as for any other programming model.

Reply:
Where does `CompilerArgs` and `LinkerArgs` come from? I mean, is it from some clang++ arguments?